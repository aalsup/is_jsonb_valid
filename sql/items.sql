-- a schema given for items
-- valid items
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '[1,2,3]');
-- wrong type of items
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '[1,"x"]');
-- ignores non-arrays
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '{"foo":"bar"}');
-- JavaScript pseudo-array is valid
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '{"0":"invalid","length":1}');
-- an array of schemas for items
-- correct types
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[1,"foo"]');
-- wrong types
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '["foo",1]');
-- incomplete array of items
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[1]');
-- array with additional items
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[1,"foo",true]');
-- empty array
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[]');
-- JavaScript pseudo-array is valid
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '{"0":"invalid","1":"valid","length":2}');