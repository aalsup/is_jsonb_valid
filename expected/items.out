-- a schema given for items
-- valid items
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '[1,2,3]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- wrong type of items
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '[1,"x"]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-arrays
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '{"foo":"bar"}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- JavaScript pseudo-array is valid
SELECT is_jsonb_valid('{"items":{"type":"integer"}}', '{"0":"invalid","length":1}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- an array of schemas for items
-- correct types
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[1,"foo"]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- wrong types
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '["foo",1]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- incomplete array of items
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[1]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- array with additional items
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[1,"foo",true]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- empty array
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '[]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- JavaScript pseudo-array is valid
SELECT is_jsonb_valid('{"items":[{"type":"integer"},{"type":"string"}]}', '{"0":"invalid","1":"valid","length":2}');
 is_jsonb_valid 
----------------
 f
(1 row)
